test/integration/scheduler/preemption_test.go

```diff
@@ -1314,3 +1314,155 @@ func TestPDBInPreemption(t *testing.T) {
 		})
 	}
 }
+
+// wrapNext wraps the NextPod() method to make it appear the preemption has been done already and the nominated node has been set.
+func wrapNext(f func() *framework.QueuedPodInfo) func() *framework.QueuedPodInfo {
+	return func() (podInfo *framework.QueuedPodInfo) {
+		podInfo = f()
+		podInfo.Pod.Status.NominatedNodeName = "node-1"
+		return podInfo
+	}
+}
+
+func initPreemptorTest(t *testing.T, nsPrefix string, opts ...scheduler.Option) *testutils.TestContext {
+	testCtx := testutils.InitTestSchedulerWithOptions(t, testutils.InitTestMaster(t, nsPrefix, nil), nil, opts...)
+	testutils.SyncInformerFactory(testCtx)
+	testCtx.Scheduler.NextPod = wrapNext(testCtx.Scheduler.NextPod)
+	go testCtx.Scheduler.Run(testCtx.Ctx)
+	return testCtx
+}
+
+// TestPreferNominatedNode test when the feature of "PreferNominatedNode" is enabled, the overall scheduling logic is not changed.
+// If the nominated node pass all the filters, then preemptor pod will run on the nominated node, otherwise, it will be scheduled
+// to another node in the cluster that ables to pass all the filters.
+// NOTE: This integration test is not intending to check the logic of preemption, but rather a sanity check when the feature is
+// enabled.
+func TestPreferNominatedNode(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PreferNominatedNode, true)()
+	testCtx := initPreemptorTest(t, "perfer-nominated-node")
+	t.Cleanup(func() {
+		testutils.CleanupTest(t, testCtx)
+	})
+	cs := testCtx.ClientSet
+	defaultPodRes := &v1.ResourceRequirements{Requests: v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewMilliQuantity(100, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI)},
+	}
+	defaultNodeRes := map[v1.ResourceName]string{
+		v1.ResourcePods:   "32",
+		v1.ResourceCPU:    "500m",
+		v1.ResourceMemory: "500",
+	}
+
+	type nodeConfig struct {
+		name string
+		res  map[v1.ResourceName]string
+	}
+
+	tests := []struct {
+		name         string
+		nodes        []*nodeConfig
+		existingPods []*v1.Pod
+		pod          *v1.Pod
+		runnningNode string
+	}{
+		{
+			name: "nominated node released all resource, preemptor is scheduled to the nominated node",
+			nodes: []*nodeConfig{
+				{name: "node-1", res: defaultNodeRes},
+				{name: "node-2", res: defaultNodeRes},
+			},
+			existingPods: []*v1.Pod{
+				initPausePod(&pausePodConfig{
+					Name:      "low-pod1",
+					Namespace: testCtx.NS.Name,
+					Priority:  &lowPriority,
+					NodeName:  "node-2",
+					Resources: defaultPodRes,
+				}),
+			},
+			pod: initPausePod(&pausePodConfig{
+				Name:      "preemptor-pod",
```

Comment:
The node1 is empty, why would this trigger preemption?

Reply:
I think we needn't test the logic of preemption, as it is already covered in the "TestPreemption", `initPreemptorTest` and `wrapNext` will set the `.Pod.Status.NominatedNodeName` for the pod so that this testcase will focus on the code change in this PR instead.

I have added some comments on this this to explain what this testcase want to cover.

Reply:
node1 is empty and `.Pod.Status.NominatedNodeName` is set to "node1", this is simulating that the resource on node1 has been released, so it's good to be assigned for the pod.

Reply:
Oh, I missed the `wrapNext`. Can you put explain the test is simulating the nominatedNode in the comment for TestPreferNominatedNode